File: backend/src/main/java/com/recipe/backend/service/RecipeService.java
package com.recipe.backend.service;

import com.recipe.backend.entity.Recipe;

import java.util.List;

public interface RecipeService {

    void loadRecipes();

    List<Recipe> getAllRecipes();

    Recipe getRecipeById(Long id);

    List<Recipe> searchRecipesByQuery(String query);
}

File: frontend/src/app/service/recipe.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Recipe } from '../model/recipe';
import { environment } from '../../environments/environment';

@Injectable({
  providedIn: 'root',
})
export class RecipeService {
  private apiUrl = environment.apiUrl + '/api/recipes';

  constructor(private http: HttpClient) {}

  getAllRecipes(): Observable<Recipe[]> {
    return this.http.get<Recipe[]>(this.apiUrl);
  }

  getRecipeById(id: number): Observable<Recipe> {
    return this.http.get<Recipe>(`${this.apiUrl}/${id}`);
  }

  searchRecipesByQuery(query: string): Observable<Recipe[]> {
    return this.http.get<Recipe[]>(`${this.apiUrl}/search`, {
      params: { query },
    });
  }
}
